lib/commands/update.js

```diff
@@ -51,51 +50,39 @@ module.exports = {
 					const stdout = response.logs.info.join( '\n' ).trim();
 
 					if ( stdout ) {
-						throw new Error( `Package "${ data.packageName }" has uncommitted changes. Aborted.` );
-					}
+						logger.error( `Package "${ data.packageName }" has uncommitted changes. Aborted.` );
 
-					return execCommand.execute( getExecData( `git fetch` ) );
-				} )
-				.then( ( response ) => {
-					log.concat( response.logs );
+						return Promise.reject();
+					}
 				} )
 				.then( () => {
-					return execCommand.execute( getExecData( `git checkout ${ data.repository.branch }` ) )
-						.catch( ( response ) => {
-							throw new Error( response.logs.error[ 0 ].replace( /^error\: /, '' ) );
-						} );
+					return execCommand.execute( getExecData( 'git fetch' ), logger );
 				} )
-				.then( ( response ) => {
-					log.concat( response.logs );
+				.then( () => {
+					return execCommand.execute( getExecData( `git checkout ${ data.repository.branch }` ), logger );
 				} )
 				.then( () => {
 					return execCommand.execute( getExecData( 'git branch' ) );
 				} )
 				.then( ( response ) => {
 					const stdout = response.logs.info.join( '\n' ).trim();
-					const isOnBranchRegexp = /HEAD detached at+/;
+					const isOnBranchRegexp = /HEAD detached at/;
 
-					// If on a detached commit, mgit must not pull the changes.
+					// If on a detached commit/tag, mgit must not pull the changes.
 					if ( isOnBranchRegexp.test( stdout ) ) {
-						log.info( `Package "${ data.packageName }" is on a detached commit.` );
+						logger.info( `Package "${ data.packageName }" is on a detached commit or tag.` );
 
-						return resolve( { logs: log.all() } );
+						return resolve( { logs: logger.all() } );
 					}
-
-					return execCommand.execute( getExecData( `git pull origin ${ data.repository.branch }` ) )
-						.catch( ( response ) => {
-							throw new Error( response.logs.error[ 0 ] );
-						} );
 				} )
-				.then( ( response ) => {
-					log.concat( response.logs );
-
-					resolve( { logs: log.all() } );
+				.then( () => {
+					return execCommand.execute( getExecData( `git pull origin ${ data.repository.branch }` ), logger );
 				} )
-				.catch( ( error ) => {
-					log.error( error );
-
-					reject( { logs: log.all() } );
+				.then( () => {
+					resolve( { logs: logger.all() } );
```

Comment:
Can't commands return the logger itself? Perhaps you could get rid of the whole logger `all()` method and the logger could be a real logger with a method to flush the logs to the std output. Right now it's just a storage, not a logger.